.github/workflows/publish-prerelease.yml

```diff
@@ -0,0 +1,72 @@
+name: Publish prerelease
+
+on:
+  workflow_call:
+    secrets:
+      PR_COMMENT_TOKEN:
+        required: true
+
+jobs:
+  publish-prerelease:
+    name: Publish prerelease
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # This is needed to get merge base to calculate bundle size diff
+
+      - name: Setup environment
+        uses: metamask/github-tools/.github/actions/setup-environment@main
+
+      - name: Get merge base commit hash
+        id: get-merge-base
+        env:
+          BASE_REF: ${{ github.event.pull_request.base.ref }}
+        run: |
+          merge_base="$(git merge-base "origin/${BASE_REF}" HEAD)"
+          echo "Merge base is '${merge_base}'"
+          echo "MERGE_BASE=${merge_base}" >> "$GITHUB_OUTPUT"
+
+      - name: Get CircleCI job details
+        id: get-circleci-job-details
+        env:
+          REPOSITORY: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+          HEAD_COMMIT_HASH: ${{ github.event.pull_request.head.sha }}
+        run: |
+          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq -r ".items | map(select(.vcs.revision == \"${HEAD_COMMIT_HASH}\" )) | first | .id")
```

Comment:
Same strategy as used elsewhere for getting the CircleCI workflow corresponding to the current branch+commit: https://github.com/MetaMask/metamask-extension/blob/6f11eda56785b6ca1bd253a6fc6a3498eef5bc5f/.github/workflows/wait-for-circleci-workflow-status.yml#L21